nipype/interfaces/dcm2nii.py

```diff
@@ -410,52 +412,44 @@ def _run_interface(self, runtime):
         # may use return code 1 despite conversion
         runtime = super(Dcm2niix, self)._run_interface(
             runtime, correct_return_codes=(0, 1, ))
-        if self.inputs.bids_format:
-            (self.output_files, self.bvecs, self.bvals,
-             self.bids) = self._parse_stdout(runtime.stdout)
-        else:
-            (self.output_files, self.bvecs, self.bvals) = self._parse_stdout(
-                runtime.stdout)
+        self._parse_files(self._parse_stdout(runtime.stdout))
         return runtime
 
     def _parse_stdout(self, stdout):
-        files = []
-        bvecs = []
-        bvals = []
-        bids = []
-        skip = False
-        find_b = False
+        filenames = []
         for line in stdout.split("\n"):
-            if not skip:
-                out_file = None
-                if line.startswith("Convert "):  # output
-                    fname = str(re.search('\S+/\S+', line).group(0))
-                    out_file = os.path.abspath(fname)
-                    # extract bvals
-                    if find_b:
-                        bvecs.append(out_file + ".bvec")
-                        bvals.append(out_file + ".bval")
-                        find_b = False
-                # next scan will have bvals/bvecs
-                elif 'DTI gradients' in line or 'DTI gradient directions' in line or 'DTI vectors' in line:
-                    find_b = True
-                if out_file:
-                    ext = '.nii' if self.inputs.compress == 'n' else '.nii.gz'
-                    files.append(out_file + ext)
-                    if self.inputs.bids_format:
-                        bids.append(out_file + ".json")
-            skip = False
-        # just return what was done
-        if not bids:
-            return files, bvecs, bvals
-        else:
-            return files, bvecs, bvals, bids
+            if line.startswith("Convert "):  # output
+                fname = str(re.search('\S+/\S+', line).group(0))
+                filenames.append(os.path.abspath(fname))
+        return filenames
+
+    def _parse_files(self, filenames):
+        outfiles, bvals, bvecs, bids = [], [], [], []
+        outtypes = (".nii", ".nii.gz", ".bval", ".bvec", ".json")
```

Comment:
@neurolabusc not that we were catching it before, but am I missing any output formats?

Reply:
The recent release (v1.0.20190410) includes ".nrrd" (`-e y -z n`) and ". nhdr"/".raw.gz" (`-e y -z y`). This was added for the Slicer team and so you may or may not want to support NRRD format export. There is also a deprecated text output  ".txt" (`-t y`) which predates BIDS.